<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ELGAMAL_UI</class>
 <widget class="QWidget" name="ELGAMAL_UI">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>442</width>
    <height>407</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_4">
   <item>
    <widget class="QPushButton" name="back_button">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="text">
      <string>返回主菜单</string>
     </property>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>密钥生成</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="generate_PG_layout">
        <item>
         <widget class="QPushButton" name="generate_PG_button">
          <property name="text">
           <string>生成&lt;p, g&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="generate_PG_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="gen_private_key_layout">
        <item>
         <widget class="QPushButton" name="gen_private_key_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>生成私钥</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="gen_private_key_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="gen_public_key_layout">
        <item>
         <widget class="QPushButton" name="gen_public_key_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>生成公钥</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="gen_public_key_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>加密</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <layout class="QHBoxLayout" name="generate_M_layout">
        <item>
         <widget class="QPushButton" name="generate_M_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>生成明文</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="generate_M_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="generate_K_layout">
        <item>
         <widget class="QPushButton" name="generate_K_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>生成整数k</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="generate_K_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="calculate_U_layout">
        <item>
         <widget class="QPushButton" name="calculate_U_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>计算整数U</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="calculate_U_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="encrypt_layout">
        <item>
         <widget class="QPushButton" name="encrypt_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>计算密文&lt;C1, C2&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="encrypt_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>解密</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <layout class="QHBoxLayout" name="calculate_V_layout">
        <item>
         <widget class="QPushButton" name="calculate_V_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>计算整数V</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="calculate_V_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QHBoxLayout" name="decrypt_layout">
        <item>
         <widget class="QPushButton" name="decrypt_button">
          <property name="enabled">
           <bool>false</bool>
          </property>
          <property name="text">
           <string>解密得到明文</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLineEdit" name="decrypt_lineEdit">
          <property name="font">
           <font>
            <pointsize>12</pointsize>
           </font>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
